cpp/modmesh/buffer/pymod/array_common.hpp

```diff
@@ -95,25 +95,30 @@ class ArrayPropertyHelper
 
         if (args.size() == 2)
         {
+            const bool isNumber = py::isinstance<py::bool_>(args[1]) || py::isinstance<py::int_>(args[1]) || py::isinstance<py::float_>(args[1]);
+
             // sarr[K] = V
-            if (py::isinstance<py::int_>(args[0]) && !py::isinstance<py::array>(args[1]))
+            if (py::isinstance<py::int_>(args[0]) && isNumber)
             {
                 const auto key = args[0].cast<ssize_t>();
 
                 arr_out.at(key) = args[1].cast<T>();
                 return;
             }
             // sarr[K1, K2, K3] = V
-            if (py::isinstance<py::tuple>(args[0]) && !py::isinstance<py::array>(args[1]))
+            if (py::isinstance<py::tuple>(args[0]) && isNumber)
             {
                 const auto key = args[0].cast<std::vector<ssize_t>>();
 
                 arr_out.at(key) = args[1].cast<T>();
                 return;
             }
+
+            const bool isArray = py::isinstance<py::list>(args[1]) || py::isinstance<py::array>(args[1]) || py::isinstance<py::tuple>(args[1]);
```

Comment:
missing support for tuple and list.

Reply:
The variable name is misleading.  Do you meant to say `is_sequence`?  See https://docs.python.org/3/c-api/sequence.html and https://docs.python.org/3/library/stdtypes.html#sequence-types-list-tuple-range .

Note that the sequence is not an iterable, which is related to [iterator](https://docs.python.org/3/c-api/iter.html).

Reply:
fixed in f9f90021927345266d1465e7c6169eeac2b58aad